File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int[] res=new int[2];
        int n=grid.length;
        int[] map=new int[(n*n)+1];

        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                map[grid[i][j]]++;
            }
        }
        for(int i=1;i<=n*n;i++){
            if(map[i]==0)res[1]=i;
            else if(map[i]==2)res[0]=i;
        }
        return res;

    }
}

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        res = []
        n = len(grid[0])
        size = n * n
        freq = [0] * (size + 1)
        a=b=0
        for r in grid:
            for num in r:
                freq[num]+=1

        for num in range(1,size+1):
            if freq[num]==2:
                a=num
            if freq[num]==0:
                b=num
        
        return [a,b]
